docs/pages/oninteractionprovider.js

```diff
@@ -0,0 +1,133 @@
+// @flow strict
+import { type Node } from 'react';
+import MainSection from '../components/MainSection.js';
+import PageHeader from '../components/PageHeader.js';
+import Page from '../components/Page.js';
+import GeneratedPropTable from '../components/GeneratedPropTable.js';
+import docgen, { type DocGen } from '../components/docgen.js';
+
+export default function OnInteractionProviderPage({
+  generatedDocGen,
+}: {|
+  generatedDocGen: DocGen,
+|}): Node {
+  return (
+    <Page title="ExperimentalOnInteractionProvider">
+      <PageHeader
+        name="ExperimentalOnInteractionProvider"
+        description={generatedDocGen?.description}
+        badge="alpha"
+      />
+      <GeneratedPropTable generatedDocGen={generatedDocGen} />
+
+      <MainSection name="Variants">
```

Comment:
Can you expand this with examples of actual usage, similar to how you expect this to be used in Pinboard? For instance: why is the component name important, and how do you expect that to be used? How would other arbitrary data passed in be used? More realistic examples than `alert`s will help communicate the intent of the component (eventually to our users, but for now just to communicate your idea for this component).